syntax = "proto2";

package ru.sktbelpa.fast_freqmeter;


enum INFO {
    // верися этого протокола
    PROTOCOL_VERSION = 0;

    // идентификатор устройства РЧ24-2.0
    FAST_AMPERMETER_ID = 0xE0F0;

    // ИД, преднозначенный для поиска
    ID_DISCOVER = 0xFFFF;

    MAGICK = 0x09;
}

//------------------------------------------------------------------------------

// Общий запрос: Вызывающий всегда посылает только этот запрос, который содержит
// все возможные данные как необязательные поля.
message Request {
    // Идентификатор запроса. Служит для определения пары запрос/ответ
    required uint32             id = 1;
    // id устройства (идентификация)
    required uint32             deviceID = 2;
    // версия протокола запрашивающего
    required uint32             protocolVersion = 3;
}

//------------------------------------------------------------------------------

// Общий ответ: Устройство всегда отвечает только этим ответом, который содержит
// все возможные данные как необязательные поля.
message Response {
    // Идентификатор запроса, которому соответствует этот ответ
    required uint32             id = 1;
    // id устройства (идентификация)
    required uint32             deviceID = 2;
    // версия протокола
    required uint32             protocolVersion = 3;
    // общий статус выполнения
    required STATUS             Global_status = 4;
}

//------------------------------------------------------------------------------

// код выполнения операции
enum STATUS {
    // успешное завершение
    OK = 0;
    // недопустимое значение поля
    ERRORS_IN_SUBCOMMANDS = 1;

    PROTOCOL_ERROR = 100;
}

//-----------------------------------------------------------------------------
